src/videojs.ads.js

```diff
@@ -147,33 +143,19 @@ var
           player.currentTime(snapshot.ended ? player.duration() : snapshot.currentTime);
         }
 
+        // Android reports duration Infinity right after switching source, so we need
+        // this to determine if ad was a postroll.
+        var androidPostroll = snapshot.ended &&
```

Comment:
Big picture, I want contrib-ads to have a clear idea of what's going on, rather than try to infer the state of things everywhere. This is one of the advantages of the architecture idea where preroll / midroll / posteroll are inherited methods: there would be no confusion about where we're at. As for a short term / incremental improvement, I'm not sure what to suggest.